### PR TITLE
[WIP/don't merge] add standalone modeling file

### PR DESCRIPTION
**Pull request by patil-suraj**
This PR adds a standalone modeling file so it would be easier for us to do modifications.

TODOs:
- [x] Support mixed precision
- [x] add activation checkpointing
- [ ] replace for loop with `jax.lax.scan` to iterate over layers for faster compile time
- [ ] MP with `pjit` (almost there!)

### CODE REVIEW

**Comment by pcuenca** [comment]
Very interesting. I didn't know about `jax.lax.scan`. One noob question: I've seen [projects use `xmap` for MP](https://github.com/kingoflolz/mesh-transformer-jax), how is that different to `pjit`?

**Comment by borisdayma** [comment]
Little comments:
* Should we call everything `DalleMiniConfig`, etc… or will we define these classes leveraging `Bart` classes like we currently have?
  * [differences with our model config](https://github.com/borisdayma/dalle-mini/blob/main/dalle_mini/model.py)
  * [setting a new config](https://github.com/borisdayma/dalle-mini/blob/db7d52106de7ff54c1019d6586df57fd2b3d566a/dev/seq2seq/run_seq2seq_flax.py#L450-L481)
* Can you add the following items and default in config
  * `image_length=256`
  * `image_vocab_size=16384`
  * `normalize_text=False`

**Comment by borisdayma** [comment]
merged through #115
